rust/connlib/snownet/src/allocation.rs

```diff
@@ -70,32 +73,40 @@ pub struct Allocation {
 /// Note that any combination of IP versions is possible here.
 /// We might have allocated an IPv6 address on a TURN server that we are talking to IPv4 and vice versa.
 #[derive(Debug, Clone, Copy)]
-pub struct Socket {
-    /// The server this socket was allocated on.
-    server: SocketAddr,
+pub struct Socket<RId> {
+    /// The ID of the relay.
+    id: RId,
```

Comment:
Can we hardcode the `Relay` type here? I don't think we will ever use something else save for testing and we can keep the types simpler.

Reply:
Unfortunately, that would create a circular dependency because it is defined in connlib, not in snownet. I tried to explain this in the doc comment of `Node`.

If I manage to make https://github.com/firezone/firezone/pull/4615/files work, then this type parameter might go away again.